### PR TITLE
Fixes exclude being out of sync in ac table

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3510
Resolves https://github.com/brave/brave-browser/issues/3524

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- clean profile
- enable rewards
- add couple sites to ac table
- remove a site from ac table
- make sure that sum is 100%
- go to the site that you removed, open panel and include it back
- make sure that site is back in ac table and sum is 100%
- make sure that exclude count is always correct

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 1]
```diff
@@ -389,6 +389,29 @@ bool PublisherInfoDatabase::RestorePublishers() {
   return restore_q.Run();
 }
 
+int PublisherInfoDatabase::GetExcludedPublishersNumber() {
```
Comment (jasonrsadler) [inline components/brave_rewards/browser/publisher_info_database.cc]: nit: Can we call `GetExcludedPublishersCount()`?

Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: sure

Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: done

**Comment by NejcZdovc** [comment]
@ryanml can you please log issue for travis problem. I think we can merge this one as other PR's has this problem as well. I run tests locally and they all pass

**Comment by NejcZdovc** [comment]
rebased to the latest master, which should have fix for travis

**Pull request merged**